src/openrct2/GameStateSnapshots.cpp

```diff
@@ -308,83 +308,82 @@ struct GameStateSnapshots : public IGameStateSnapshots
     }
 
     void CompareSpriteDataVehicle(
-        const rct_vehicle& spriteBase, const rct_vehicle& spriteCmp, GameStateSpriteChange_t& changeData) const
+        const Vehicle& spriteBase, const Vehicle& spriteCmp, GameStateSpriteChange_t& changeData) const
     {
-        COMPARE_FIELD(rct_vehicle, vehicle_sprite_type);
-        COMPARE_FIELD(rct_vehicle, bank_rotation);
-        COMPARE_FIELD(rct_vehicle, remaining_distance);
-        COMPARE_FIELD(rct_vehicle, velocity);
-        COMPARE_FIELD(rct_vehicle, acceleration);
-        COMPARE_FIELD(rct_vehicle, ride);
-        COMPARE_FIELD(rct_vehicle, vehicle_type);
-        COMPARE_FIELD(rct_vehicle, colours);
-        COMPARE_FIELD(rct_vehicle, track_progress);
-        COMPARE_FIELD(rct_vehicle, track_direction);
-        COMPARE_FIELD(rct_vehicle, track_x);
-        COMPARE_FIELD(rct_vehicle, track_y);
-        COMPARE_FIELD(rct_vehicle, track_z);
-        COMPARE_FIELD(rct_vehicle, next_vehicle_on_train);
-        COMPARE_FIELD(rct_vehicle, prev_vehicle_on_ride);
-        COMPARE_FIELD(rct_vehicle, next_vehicle_on_ride);
-        COMPARE_FIELD(rct_vehicle, var_44);
-        COMPARE_FIELD(rct_vehicle, mass);
-        COMPARE_FIELD(rct_vehicle, update_flags);
-        COMPARE_FIELD(rct_vehicle, swing_sprite);
-        COMPARE_FIELD(rct_vehicle, current_station);
-        COMPARE_FIELD(rct_vehicle, swinging_car_var_0);
-        COMPARE_FIELD(rct_vehicle, var_4E);
-        COMPARE_FIELD(rct_vehicle, status);
-        COMPARE_FIELD(rct_vehicle, sub_state);
+        COMPARE_FIELD(Vehicle, vehicle_sprite_type);
+        COMPARE_FIELD(Vehicle, bank_rotation);
+        COMPARE_FIELD(Vehicle, remaining_distance);
+        COMPARE_FIELD(Vehicle, velocity);
+        COMPARE_FIELD(Vehicle, acceleration);
+        COMPARE_FIELD(Vehicle, ride);
+        COMPARE_FIELD(Vehicle, vehicle_type);
+        COMPARE_FIELD(Vehicle, colours);
+        COMPARE_FIELD(Vehicle, track_progress);
+        COMPARE_FIELD(Vehicle, track_direction);
+        COMPARE_FIELD(Vehicle, track_x);
+        COMPARE_FIELD(Vehicle, track_y);
+        COMPARE_FIELD(Vehicle, track_z);
+        COMPARE_FIELD(Vehicle, next_vehicle_on_train);
+        COMPARE_FIELD(Vehicle, prev_vehicle_on_ride);
+        COMPARE_FIELD(Vehicle, next_vehicle_on_ride);
+        COMPARE_FIELD(Vehicle, var_44);
+        COMPARE_FIELD(Vehicle, mass);
+        COMPARE_FIELD(Vehicle, update_flags);
+        COMPARE_FIELD(Vehicle, swing_sprite);
+        COMPARE_FIELD(Vehicle, current_station);
+        COMPARE_FIELD(Vehicle, swinging_car_var_0);
+        COMPARE_FIELD(Vehicle, var_4E);
+        COMPARE_FIELD(Vehicle, status);
+        COMPARE_FIELD(Vehicle, sub_state);
         for (int i = 0; i < 32; i++)
```

Comment:
While we're at it, could you get rid of these hard coded maximums?

Reply:
I'll address that in a follow up pr.